src/CareTogether.Core/Utilities/AppendBlobMultitenantEventLog.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Newtonsoft.Json;
+using OneOf;
+using OneOf.Types;
+
+
+namespace CareTogether.Utilities
+{
+    public class AppendBlobMultitenantEventLog<T>
+    {
+        private LogType _logType;
+        private BlobServiceClient _blobServiceClient;
+        private ConcurrentDictionary<(Guid organizationId, Guid locationId), int> _tenantEventCounts;
```

Comment:
This will have an initial value of `null`, so you need to initialize it (either here or in the constructor) with a `new ConcurrentDictionary<(Guid organizationId, Guid locationId), int>`.